src/kayenta/report/detail/detail.tsx

```diff
@@ -4,13 +4,46 @@ import ReportHeader from './header';
 import ReportScores from './reportScores';
 import MetricResults from './metricResults';
 
+import './detail.less';
+
 /*
 * Layout for report detail view.
 * */
-export default () => (
-  <div className="vertical flex-1">
-    <ReportHeader />
-    <ReportScores />
-    <MetricResults />
-  </div>
-);
+
+export interface IDetailViewStateProps {
+  isExpanded: boolean;
+}
+
+export default class DetailView extends React.Component<any, IDetailViewStateProps> {
+  constructor(props: any) {
+    super(props);
+    this.state = { isExpanded: true };
+  }
+
+  private toggleDetailHeader = (): void => {
+    this.setState({ isExpanded: !this.state.isExpanded });
```

Comment:
Could you please use the redux store to toggle the header instead of using local state? deck-kayenta has been very consistent about this and I think it's a good convention to keep.

Reply:
I'm not sure this is a productive constraint to hold onto as we continue to increase the complexity of the UI and add more ephemeral state. I'm a huge fan of redux, and fully support keeping all cross-component state restricted to the store. That being said, I don't think it's useful to lift up state that's purely presentational/ephemeral, doesn't touch any components above or below the local component in the tree, and doesn't depend on any state from the store. The moment this state became anything more concrete or externally visible outside of its component I'd likely change my view.